File: Rivington.IG.Web/Rivington.IG.Domain.Models/OrderManagement/InspectionOrderProperty/HighValueKitchen.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using Microsoft.EntityFrameworkCore;

namespace Rivington.IG.Domain.Models.OrderManagement
{
    public class InspectionOrderPropertyHighValueKitchen
    {
        [Parent]
        public InspectionOrderProperty InspectionOrderProperty { get; set; }
        [Key]
        public Guid Id { get; set; }
        

        public string KitchenCabinetId { get; set; }
        [ForeignKey(nameof(KitchenCabinetId))]
        public KitchenCabinet KitchenCabinet { get; set; }

        public string KitchenCountertopId { get; set; }
        [ForeignKey(nameof(KitchenCountertopId))]
        public KitchenCountertop KitchenCountertop { get; set; }

        public string KitchenBacksplashMaterial { get; set; }
        public bool KitchenIsland { get; set; }
        public string IslandCabinetMaterial { get; set; }
        public string IslandCountertopMaterial { get; set; }

        /// <summary>
        /// easy but not pretty implementation
        /// saving of list is comma delimited
        /// </summary>
        //[NotMapped]
        //public ICollection<string> ApplianceTypeList { get; set; }
        //[Column]
        //private string ApplianceType
        //{
        //    get => string.Join(",", ApplianceTypeList);
        //    set => ApplianceTypeList = value.Split(',').ToList();
        //}

        public ICollection<InspectionOrderPropertyHighValueKitchenAppliances> Appliances { get; set; }

        public static void BuildModel(ModelBuilder modelBuilder)
        {
            var entTypeName = nameof(InspectionOrderPropertyHighValueKitchen);
            modelBuilder.Entity<InspectionOrderProperty>()
                .HasOne(a => a.HighValueKitchen).WithOne(b => b.InspectionOrderProperty)
                .HasForeignKey<InspectionOrderPropertyHighValueKitchen>()
                .HasConstraintName($"FK_{nameof(InspectionOrderProperty)}_{entTypeName}")
                .OnDelete(DeleteBehavior.Cascade);

            var entBuilder = modelBuilder.Entity<InspectionOrderPropertyHighValueKitchen>();
            entBuilder.HasOne(a => a.KitchenCabinet).WithMany().HasConstraintName($"FK_{entTypeName}_{nameof(KitchenCabinet)}");
            entBuilder.HasOne(a => a.KitchenCountertop).WithMany().HasConstraintName($"FK_{entTypeName}_{nameof(KitchenCountertop)}");

            InspectionOrderPropertyHighValueKitchenAppliances.BuildModel(modelBuilder);
        }
    }
}

File: Rivington.IG.Web/Rivington.IG.API/src/app/modules/shared/models/ordermanagement/inspection-order/checklist/property/high-value-kitchen.ts
import { InspectionOrderProperty } from "../property";
import { InspectionOrderPropertyHighValueKitchenAppliances } from "./high-value-kitchen/InspectionOrderPropertyHighValueKitchenAppliances";
import { ApplianceBrand } from "./high-value-kitchen/appliance-brand";
import { KitchenCabinet } from "./high-value-kitchen/kitchen-cabinet";
import { KitchenCountertop } from "./high-value-kitchen/kitchen-countertop";

export class InspectionOrderPropertyHighValueKitchen {
    /*[Parent]*/
    inspectionOrderProperty?: InspectionOrderProperty;
    /*[Key]*/
    id?: string;
    kitchenCabinetId?: string;
    /*[ForeignKey(nameof(KitchenCabinetId))]*/
    kitchenCabinet?: KitchenCabinet;
    kitchenCountertopId?: string;
    /*[ForeignKey(nameof(KitchenCountertopId))]*/
    kitchenCountertop?: KitchenCountertop;
    kitchenBacksplashMaterial?: string;
    kitchenIsland?: boolean;
    islandCabinetMaterial?: string;
    islandCountertopMaterial?: string;
    appliances?: InspectionOrderPropertyHighValueKitchenAppliances[];
    public constructor(init?:Partial<InspectionOrderPropertyHighValueKitchen>) {
        Object.assign(this, init);
    }
  }